decryptor/postgresql/pg_decryptor.go

```diff
@@ -205,15 +205,15 @@ func checkInlinePoisonRecordInBlock(block []byte, decryptor base.Decryptor, logg
 		logger.Debugln("Check poison records")
 		currentIndex := 0
 		for {
-			if index, _ := decryptor.BeginTagIndex(block[currentIndex:]); index == utils.NotFound {
-				return nil
-			} else {
+			if index, _ := decryptor.BeginTagIndex(block[currentIndex:]); index != utils.NotFound {
```

Comment:
i would change it

```
if ... {
  return nil
}
...
currentIndex++
```

Reply:
we need `index` value that added to `currentIndex`. Now it in `if expression`. To do like you want we need to move assignment from if block to outer. then we can use this value
```
index, _ := decryptor.BeginTagIndex(block[currentIndex:])
if index == utils.NotFound {
   return nil 
}
currentIndex += index
.....
```
this variant is better?

Reply:
yes, i prefer this updated version